build/__tools__/run-benchmarks.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+SCRIPTDIR="$(dirname $(readlink --canonicalize ${BASH_SOURCE}))"
+SRCDIR="$(realpath ${SCRIPTDIR}/../..)"
+
+source ${SRCDIR}/bin/lib/common.sh
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+if [ "${PDO_INTERPRETER}" == "gipsy" ]; then
+    die automated benchmarks not enabled for the gipsy interpreter
+fi
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+PY3_VERSION=$(python --version | sed 's/Python 3\.\([0-9]\).*/\1/')
+if [[ $PY3_VERSION -lt 5 ]]; then
+    die activate python3 first
+fi
+
+: "${PDO_SOURCE_ROOT:-$(die Missing environment variable PDO_SOURCE_ROOT)}"
+: "${PDO_INTERPRETER:-$(die Missing environment variable PDO_INTERPRETER)}"
+
+# wawaka contracts reside under same directory, regardless of execution mode
+PDO_INTERPRETER_NAME=${PDO_INTERPRETER}
+if [[ "$PDO_INTERPRETER" =~ ^"wawaka-" ]]; then
+    PDO_INTERPRETER_NAME="wawaka"
+fi
+
+# build interpreter specific benchmark contracts
+yell --------------- Building benchmark contracts ---------------
+cd $SRCDIR/contracts/${PDO_INTERPRETER_NAME}/benchmarks
+try make all
+try make install
+
+# execute interpreter specific benchmarks
+yell --------------- Executing benchmarks ---------------
+for test_file in ${PDO_SOURCE_ROOT}/build/benchmarks/${PDO_INTERPRETER_NAME}/*.json ; do
+    test_contract=$(basename ${test_file} .json)
```

Comment:
you might want to explicitly mention the list of tests done rather than *.json. it might be a good idea going forward to be able to add more tests to this folder, but not necessarily run with this script.

Reply:
What do we gain by calling the explicit tests in this script? This script mirrors the existing `run-tests.sh` script; just like with the regular tests, if someone wants to run an individual test, they can do so by calling the `pdo-test-contract` command for the specific test.